git/git-plan.md

```diff
@@ -0,0 +1,88 @@
+# Version Control with Git & GitHub
+
+## Introduction
+
+- what is version control?
+- why should I use it?
+- what is git?
+- what is GitHub?
+
+## GitHub
+
+- look at an example repo
+    - e.g. this one? One of the Software Carpentry lessons? any other suggestions?
+    - basic structure of 'front page' - list of files, README rendered, commit messages, last modified time, contributors
+    - look at graph of history, branches, commit messages and codes
+    - open an example file, use blame to look at when a line was added/most recently changed, and by whom
+- create your own repository
+    - look at README
+    - edit README, quick intro to Markdown
+    - commit messages - what they are; advice for writing good ones
+    - look at history
+    - repeat steps above
+    - add a new file (python script), commit, look at result
+- discuss limitations of online-only approach
+    - how can we be sure that our script is doing what we intend?
+    - would like to make changes, test, then update the material online when we're sure that it's ready
+- now switch to the command line
+
+## Command Line
+
+- navigate to relevant directory
+- clone repo
+- change global settings
+  - `git config --global user.name`
+  - `git config --global user.email`
+  - `git config --global core.editor "nano -w"`
+- mention `git init`
+- `git status`
+- `git add`
+- `git commit`
+  - commit messages
+  - imagine your future self as a collaborator, who won't know (remember) why you made the changes you're making
+- `git add`
+- `git commit -m`
+- `git commit -a`
```

Comment:
Not a big fan of `git commit -a` as it encourages committing without thinking what to commit. IMHO staging is an important step.

Reply:
Very valid point. Happy to leave this out.